src/insertWord.js

```diff
@@ -17,3 +18,28 @@ insertUi.addWordBtn.addEventListener("click", () => {
 
   insertUi.clear();
 });
+
+insertUi.doneBtn.addEventListener('click', (e) => {
+  if (addWordSuccess) {
+    const word = insertUi.createWordForUser(JSON.parse(localStorage.getItem(LOCALSTORAGE_KEY)));
+    wordService.saveWord(word);
+  
+    insertUi.clear();
+    addWordSuccess = null;
+  } else if (addWordSuccess === null) {
+    e.preventDefault();
+    const inputPass = insertUi.checkInputs();
+    if (!inputPass) {
+      insertUi.showAlert("Please fill the word input and make a recording or fill in the definition or example sentence input", "danger");
+      addWordSuccess = false;
+      insertUi.doneBtn.click();
+      return;
+    }
+    insertUi.doneBtn.click();
+    addWordSuccess = true;
+  } else if (!addWordSuccess) {
+    e.preventDefault();
+    addWordSuccess = null;
+  }
```

Comment:
1. the `else if (! addWordSuccess)` is not necessary, you are supposed to use only `else` to catch all remaining possible scenarios.
2. this part of logic is not necessary as the above comments